rust/operator-binary/src/controller.rs

```diff
@@ -960,31 +830,31 @@ fn env_var_from_secret(secret_name: &Option<String>, env_var: &str) -> Option<En
 /// * `product_config`  - The product config to validate and complement the user config.
 ///
 fn validated_product_config(
-    resource: &TrinoCluster,
+    trino: &TrinoCluster,
     version: &str,
     product_config: &ProductConfigManager,
 ) -> Result<ValidatedRoleConfigByPropertyKind, Error> {
     let mut roles = HashMap::new();
 
-    let config_files = vec![
+    let mut config_files = vec![
         PropertyNameKind::File(CONFIG_PROPERTIES.to_string()),
         PropertyNameKind::File(HIVE_PROPERTIES.to_string()),
         PropertyNameKind::File(NODE_PROPERTIES.to_string()),
         PropertyNameKind::File(JVM_CONFIG.to_string()),
         PropertyNameKind::File(LOG_PROPERTIES.to_string()),
     ];
 
+    if trino.get_authentication().is_some() {
+        config_files.push(PropertyNameKind::File(
+            PASSWORD_AUTHENTICATOR_PROPERTIES.to_string(),
```

Comment:
Is there a problem with including this file if authn is disabled?

Reply:
I dont think so but why?

Reply:
Less code, fewer special cases.